compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -1120,7 +1120,14 @@ object Semantic {
 
       case Call(ref, argss) =>
         // check args
-        val (errors, args) = evalArgs(argss.flatten, thisV, klass)
+        val (argErrors, args) = evalArgs(argss.flatten, thisV, klass)
+        // Allow cold args for static methods with non-matchable params
+        val methodType = ref.symbol.info.stripPoly
+        val allMatchable = methodType.paramInfoss.flatten.forall { (info) => info <:< defn.MatchableType }
```

Comment:
I think we wanted to do the check per-argument rather than for all the arguments. That is, in `def call`, we have
```scala
      def checkArgs = args.flatMap(_.promote)
```
There we want to promote only those args that correspond to non-matchable params. Even if some params are matchable, the call is still fine as long as we can promote the args that correspond to those params.